Sources/TuistCore/Cache/CacheDirectoriesProvider.swift

```diff
@@ -38,39 +20,26 @@ public final class CacheDirectoriesProvider: CacheDirectoriesProviding {
         }
     }
 
-    public var pluginCacheDirectory: AbsolutePath {
-        cacheDirectory.appending(component: "Plugins")
-    }
-
-    public var testsCacheDirectory: AbsolutePath {
-        cacheDirectory.appending(component: "TestsCache")
-    }
-
-    public var buildCacheDirectory: AbsolutePath {
-        cacheDirectory.appending(component: "BuildCache")
-    }
-
-    public var generatedAutomationProjectsDirectory: AbsolutePath {
-        cacheDirectory.appending(component: "Projects")
-    }
-
-    public var projectDescriptionHelpersCacheDirectory: AbsolutePath {
-        cacheDirectory.appending(component: "ProjectDescriptionHelpers")
-    }
-
-    public var manifestCacheDirectory: AbsolutePath {
-        cacheDirectory.appending(component: "Manifests")
+    public func cacheDirectory(for category: CacheCategory) -> AbsolutePath {
+        cacheDirectory.appending(component: category.directoryName)
     }
+}
 
-    public var cacheDirectories: [AbsolutePath] {
-        [
-            cacheDirectory,
-            pluginCacheDirectory,
-            buildCacheDirectory,
-            testsCacheDirectory,
-            generatedAutomationProjectsDirectory,
-            projectDescriptionHelpersCacheDirectory,
-            manifestCacheDirectory,
-        ]
+extension CacheCategory {
+    var directoryName: String {
+        switch self {
+        case .plugins:
+            return "Plugins"
+        case .builds:
+            return "BuildCache"
+        case .tests:
+            return "TestsCache"
```

Comment:
Maybe for 2.0 we could rename these to Builds and Tests, so it matches the other directories?

Reply:
Yep, I agree it would makes sense, let's see what @tuist/core thinks

Reply:
I think that makes sense as well 👍